src/dos/dos_keyboard_layout.cpp

```diff
@@ -922,62 +920,69 @@ Bitu keyboard_layout::read_codepage_file(const char* codepage_file_name, Bit32s
 		return KEYB_INVALIDCPFILE;
 	}
 
-	number_of_codepages=host_readw(&cpi_buf[start_pos]);
+	const auto number_of_codepages = host_readw(&cpi_buf.at(start_pos));
 	start_pos+=4;
 
 	// search if codepage is provided by file
-	for (Bit16u test_codepage=0; test_codepage<number_of_codepages; test_codepage++) {
-		Bit16u device_type, font_codepage, font_type;
-
+	for (uint16_t test_codepage = 0; test_codepage < number_of_codepages; test_codepage++) {
 		// device type can be display/printer (only the first is supported)
-		device_type=host_readw(&cpi_buf[start_pos+0x04]);
-		font_codepage=host_readw(&cpi_buf[start_pos+0x0e]);
+		const auto device_type_index = start_pos + 0x04;
```

Comment:
For all of these `_index` variables, I'd be inclined to make the hex values meaningful instead. Because the variables are kinda... redundant otherwise.

Reply:
Yes, that will be clearer. 
The hex values are all offsets relative to a prior field, so can store the offsets as named constexpr's , and then build up the absolute indexes from the offsets.

Reply:
I couldn't make them any more meaningful.. the hex values are fixed offsets from wherever the start of the font segment is found. 

Switched back to the more concise "start_pos + hex-offset"; it's self-documenting enough given the variable name indicates what's being stored (and also describes the offset). 

```c++
const auto device_type = host_readw(&cpi_buf.at(start_pos + 0x04));
const auto font_codepage = host_readw(&cpi_buf.at(start_pos + 0x0e));
const auto font_data_header_pt = host_readd(&cpi_buf.at(start_pos + 0x16));
```